daemons/pacemakerd/pacemaker.service.in

```diff
@@ -73,14 +73,15 @@ SendSIGKILL=no
 #
 # ExecStopPost=/bin/sh -c 'pidof pacemaker-controld || killall -TERM corosync'
 
-# Uncomment this for older versions of systemd that didn't support
-# TimeoutStopSec
-# TimeoutSec=30min
-
-# Pacemaker can only exit after all managed services have shut down
-# A HA database could conceivably take even longer than this 
+# Free of faults, Pacemaker can only exit after all managed services have shut
+# down, so be advised an HA database could conceivably take a very long time.
+# Also, in some corner cases, shutdown can be blocked from within the Pacemaker
+# on purpose for a period set with "shutdown-escalation" cluster option (20 min
+# by default) so ensure the stop timeout assigned below shows more generosity.
```

Comment:
Adapting my previous suggestion, "A properly functioning Pacemaker service will wait to exit until all resources have been stopped on the local node, and potentially across all nodes that are shutting down. The default TimeoutStopSec of 30min should cover most typical cluster configurations, but it may need adjusting for local conditions (for example, a large, clustered database could conceivably take longer to stop)."

I don't think we need to mention shutdown-escalation. It doesn't block shutdown, but rather causes the DC to release the DC role and exit if shutdown takes longer than that. It's primarily a failsafe in case there's a failed transition (e.g. graph loop) at shutdown. I'm not sure if it imposes another limit on how long resource stops can take, but quite possibly -- if so, we should document that, but if that's the case I'm surprised it's documented as an advanced option.

Reply:
> I don't think we need to mention shutdown-escalation. It doesn't block
> shutdown

Problem is that it apparently does in some corner cases, which led me to
this PR in the first place.  Currently, the suggested inequation holds
(not sure if purposefully or accidentally), but the user can deliberately
violate that, effectively handing over the terminator role to systemd,
which implies inability to do the right thing on the pacemaker side
(proper reporting about the situation arisen at the very least).
So better to suggest against the such violation, and we cannot then
make do without mentioning the other configurable it relates to.

I wouldn't open this PR just for the sake of rehashing the existing
wording.

Reply:
Relevant records I've dug up:
https://developerbugs.linuxfoundation.org/show_bug.cgi?id=146523
ec4c1206d80e373c2ee53081009012d932d0bd81
980a5df07672ac4ad9055bc288665f18cd508a18

Please, do your checking when claiming something, otherwise just
an energy burning noise gets spread.

Reply:
> Problem is that it apparently does in some corner cases, which led me to
> this PR in the first place.

I think we're just using different terminology: shutdown is never blocked *on purpose*, and shutdown-escalation *ends* the block rather than causing it. If clean shutdown of the controller is blocked due to some complication (such as a graph loop in the shutdown transition), the controller will be unable to stop. When the shutdown escalation timer pops in that situation, the controller will immediately release the DC role and exit uncleanly.

The shutdown-escalation route is slightly cleaner than TimeoutStopSec because it allows the other daemons to exit normally, but that's of minor benefit -- the controller exits uncleanly in either case, and that's the main cause of shutdown problems. I understood what you were trying to say, I'm just not sure whether it's worth drawing attention to shutdown-escalation for that difference.

If shutdown-escalation serves as a cap on how long the controller will wait for resources to cleanly stop (I suspect it does but haven't tested), then we should update both its documentation in Pacemaker Explained and this here to explain how the two combine. In other words, I want to see if shutdown-escalation applies when resource stops are proceeding normally but taking a long time (vs when the controller is having problems, where it definitely applies). If it does apply to resource stops, then yes, I do want to document it here.

> https://developerbugs.linuxfoundation.org/show_bug.cgi?id=146523

It says bug does not exist, is the link cut off?